src/actions/result.ts

```diff
@@ -18,22 +20,26 @@ export type ResultRequest = ReduxAction<typeof RESULT_REQUEST, {}>;
 
 export const RESULT_RECEIVE = 'RESULT_RECEIVE';
 export type ResultReceive = ReduxAction<typeof RESULT_RECEIVE, {
-  modelGroup: SpecQueryModelGroup
+  modelGroup: SpecQueryGroup<PlotObject>
 }>;
 
 export type AsyncResultRequest = ThunkAction<void , State, undefined>;
-export function resultRequest(query?: Query, schema?: Schema): AsyncResultRequest {
+export function resultRequest(query?: Query, schema?: Schema, data?: Data): AsyncResultRequest {
   return (dispatch: Dispatch<Action>, getState) => {
     if (!query) {
       query = getQuery(getState());
     }
     if (!schema) {
       schema = getSchema(getState());
     }
+    if (!data) {
```

Comment:
It seems like the input  `query`, `schema`, `data` are actually ignored and override

I wonder if we only need only one between (1) `query`, `schema`, `data` as parameters of `resultRequest`  or (2) `getState` as a parameter of the returned method.